ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/discovery/PeerDiscoveryAgent.java

```diff
@@ -150,7 +150,7 @@ public PeerDiscoveryAgent(
       LOG.info("Starting peer discovery agent on host={}, port={}", host, port);
 
       vertx
-          .createDatagramSocket(new DatagramSocketOptions())
+          .createDatagramSocket(new DatagramSocketOptions().setIpV6(true))
```

Comment:
Have you tested this on a system with no IPv6 support at all?  There's a lot of paths things could go down but if it goes down the Java NIO datagram support route (which is the only place it makes a difference) it will throw `UnsupportedOperationException` if IPv6 is not available.

Also, is there any indication that this is required?  Have we seen it not work on an IPv6 only network?

Reply:
Our jenkins build system is ipv4 only, and yes, it broke there.